### PR TITLE
Add leader election

### PR DESCRIPTION
**Pull request by fisherxu**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/mpi-operator/110)

### CODE REVIEW

File: cmd/mpi-operator.v1alpha2/app/server.go [hunk 1]
```diff
@@ -67,49 +93,138 @@ func Run(opt *options.ServerOption) error {
...
+		}
+	}
+
+	id, err := os.Hostname()
 	if err != nil {
-		glog.Fatalf("Error building kubeflow clientset: %s", err.Error())
+		return fmt.Errorf("failed to get hostname: %v", err)
+	}
+	// add a uniquifier so that two processes on the same host don't accidentally both become active
+	id = id + "_" + string(uuid.NewUUID())
```
Comment (fisherxu) [inline cmd/mpi-operator.v1alpha2/app/server.go]: if don't add uuid to id, when the host has same hostname, the process will all become active.

**Comment by fisherxu** [comment]
/assign @rongou @terrytangyuan 

Just add K8s leader election code, and have tested, ptal.

**Comment by fisherxu** [comment]
> Should do the similar change to tf and pytorch?

add uuid to id? @gaocegege

**Comment by gaocegege** [comment]
@fisherxu Yeah. I think if mpi operator has the problem, then tf-operator and pytorch-operator will have the same problem, too.

**Pull request merged**